py/client/pydeephaven/ticket.py

```diff
@@ -145,11 +145,11 @@ def __init__(self, ticket_bytes: bytes):
             raise DHError('ApplicationTicket: ticket is None')
         elif not ticket_bytes.startswith(b'a/'):
             raise DHError(f'ApplicationTicket: ticket {ticket_bytes} is not an application ticket')
-        elif len(ticket_bytes.split(b'/')) != 3:
+        elif len(ticket_bytes.split(b'/')) != 4:
```

Comment:
These seem like kind of serious bugs! How did we not catch them in a unit test? I'll note that I don't see new unit tests being added.

Reply:
Setting up app mode in the server requires quite a bit of work. That's why we don't CI test the app mode for the server API either. Maybe a ticket for future improvement?